clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapFactory.java

```diff
@@ -118,7 +118,7 @@ static CoreState getCoreState(Context context, CleverTapInstanceConfig cleverTap
                 ctLockManager, callbackManager, deviceInfo, baseDatabaseManager);
         coreState.setControllerManager(controllerManager);
 
-        TriggersMatcher triggersMatcher = new TriggersMatcher();
+        TriggersMatcher triggersMatcher = new TriggersMatcher(localDataStore);
```

Comment:
can we avoid passing the entire localdatastore. 
Ideally we should avoid database queries while triggers matching as it will slow down sdk. Also please pass a closure of only things needed to be done instead of entire class so this is testable

Reply:
It's required as per logic.

Reply:
Using a Set() as a caching mechanism.